internal/sdkprovider/service/organization/organization_user_data_source.go

```diff
@@ -6,25 +6,96 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
+	"github.com/aiven/aiven-go-client/v2"
 	"github.com/aiven/terraform-provider-aiven/internal/schemautil"
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil/userconfig"
 )
 
 func DatasourceOrganizationUser() *schema.Resource {
 	return &schema.Resource{
 		ReadContext: datasourceOrganizationUserRead,
 		Description: "The Organization User data source provides information about the existing Aiven" +
 			" Organization User.",
-		Schema: schemautil.ResourceSchemaAsDatasourceSchema(
-			aivenOrganizationUserSchema, "organization_id", "user_email",
-		),
+		Schema: map[string]*schema.Schema{
+			"organization_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: userconfig.Desc("The unique organization ID").Build(),
+			},
+			"user_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The unique organization user ID",
+			},
+			"user_email": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "This is a user email address",
+			},
+			"create_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Time of creation",
+			},
+		},
 	}
 }
 
+// datasourceOrganizationUserRead reads the specified Organization User data source.
 func datasourceOrganizationUserRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	organizationID := d.Get("organization_id").(string)
 	userEmail := d.Get("user_email").(string)
+	userId := d.Get("user_id").(string)
+
+	if userEmail == "" && userId == "" {
+		return diag.Errorf("either user_email or user_id must be specified")
+	}
+
+	client := m.(*aiven.Client)
+	rm, err := client.OrganizationUser.List(ctx, organizationID)
+	if err != nil {
+		return diag.Errorf("cannot get organization [%s] user list: %s", organizationID, err)
+	}
+
+	var found int
+
+	for _, user := range rm.Users {
+		var userInfo *aiven.OrganizationUserInfo
+
+		if userEmail != "" && user.UserInfo.UserEmail == userEmail {
+			userInfo = &user.UserInfo
+		}
+
+		if userId != "" && user.UserID == userId {
+			userInfo = &user.UserInfo
+		}
+
+		if userInfo != nil {
+			found++
+			if err := d.Set("organization_id", organizationID); err != nil {
+				return diag.FromErr(err)
+			}
+			if err := d.Set("user_email", userInfo.UserEmail); err != nil {
+				return diag.FromErr(err)
+			}
+			if err := d.Set("create_time", user.JoinTime.String()); err != nil {
+				return diag.FromErr(err)
+			}
+			if err := d.Set("user_id", user.UserID); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	if found == 0 {
+		return diag.Errorf("organization user %s not found in organization %s", userEmail, organizationID)
+	}
+
+	if found > 1 {
+		return diag.Errorf("multiple organization users %s found in organization %s", userEmail, organizationID)
+	}
```

Comment:
nit: I would redesign this a bit just not to make grow the part with the loop with more "set"s when new fields added. Both id and email are [mandatory](https://api.aiven.io/doc/#tag/Users/operation/OrganizationUserList), so you can omit empty checks.

```go
        var found int
	var userInfo *aiven.OrganizationUserInfo
	for _, user := range rm.Users {
		if user.UserID == userId || user.UserInfo.UserEmail == userEmail {
			userInfo = &user.UserInfo
          		found++
		}
	}

	if found != 1 {
		return diag.Errorf("expected exactly one user with email %q or id=%q, found %d", userEmail, userId, found)
	}

	if err := d.Set("organization_id", organizationID); err != nil {
		return diag.FromErr(err)
	}
	if err := d.Set("user_email", userInfo.UserEmail); err != nil {
		return diag.FromErr(err)
	}
	if err := d.Set("create_time", user.JoinTime.String()); err != nil {
		return diag.FromErr(err)
	}
	if err := d.Set("user_id", user.UserID); err != nil {
		return diag.FromErr(err)
	}
        // this part can grow now
```

Reply:
Updated code to something similar